crates/sui/src/client_commands.rs

```diff
@@ -1282,128 +1136,43 @@ impl SuiClientCommands {
                     .move_call_tx_kind(package, &module, &function, type_args, args)
                     .await?;
 
-                let gas_owner = context.try_get_object_owner(&gas).await?;
-                let signer = gas_owner.unwrap_or(context.active_address()?);
-                let gas_price = if let Some(gas_price) = gas_price {
-                    gas_price
-                } else {
-                    context.get_reference_gas_price().await?
-                };
+                let sender = context.try_get_object_owner(&opts.gas).await?;
+                let sender = sender.unwrap_or(context.active_address()?);
```

Comment:
```suggestion
                let sender = if let Some(sender) = sender {
                    sender
                } else {
                    context.active_address()?
                }
```

Avoid the computation if we have a sender.